### ISSUE TITLE
Crash on trying to open handbook page for some items

### ISSUE THREAD

**Issue by Craluminum2413** [anegostudios/VintageStory-Issues#1430]
**Game Version:** 1.16.0-pre.1  
**Platform:** Windows
**Modded:** No
**SP/MP:** Singleplayer

### Description
It is crashing for:
- Beeswax
- Honeycomb (someone reported, but I cannot reproduce that)

### How to reproduce
1. Get beeswax
2. Press H to open its handbook page
3. It crashes

### Logs
```
Version: v1.16.0-pre.1 (Unstable)12/01/2021 14:42:35: Critical error occurred
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.API.Common.CollectibleObject.UpdateAndGetTransitionStatesNative(IWorldAccessor world, ItemSlot inslot)
   at Vintagestory.API.Common.CollectibleObject.UpdateAndGetTransitionState(IWorldAccessor world, ItemSlot inslot, EnumTransitionType type)
   at Vintagestory.Client.NoObf.InventoryItemRenderer.GetItemStackRenderInfo(ClientMain game, ItemSlot inSlot, EnumItemRenderTarget target)
   at Vintagestory.Client.NoObf.InventoryItemRenderer.RenderItemstackToGui(ItemSlot inSlot, Double posX, Double posY, Double posZ, Single size, Int32 color, Boolean shading, Boolean origRotate, Boolean showStackSize)
   at Vintagestory.API.Client.SlideshowItemstackTextComponent.RenderInteractiveElements(Single deltaTime, Double renderX, Double renderY)
   at Vintagestory.API.Client.GuiElementRichtext.RenderInteractiveElements(Single deltaTime)
   at Vintagestory.API.Client.GuiComposer.Render(Single deltaTime)
   at Vintagestory.API.Client.GuiDialog.OnRenderGUI(Single deltaTime)
   at Vintagestory.Client.NoObf.GuiManager.OnRenderFrameGUI(Single deltaTime)
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt)
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt)
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt)
   at _BUMabDMvB5g1D4cwmJFdUV7RB7B._nA3yaaylsiYnA09wfmcvIIKlcbN(Single )
   at _SGHyUxV5Rv3GnxQS6X37QfGaYYI._wtaWT1dJOq4AyHDjHsCHfaFTnkQ(Single )
   at _SGHyUxV5Rv3GnxQS6X37QfGaYYI._t0woIE9wg5YcVEaBBPQohFJK0os(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp) in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\GameWindow.cs:line 479
   at OpenTK.GameWindow.DispatchRenderFrame() in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\GameWindow.cs:line 454
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\GameWindow.cs:line 375
   at _qGIeLvQF356cX6MlSgQecVUPPnB._1UOaKCATq4VGq5WIs6BsnCl9AiN(_hepllT23IKsNQYqswdyIdzYZoCd , String[] )
   at _Va34feq3eOguuxZWZKrRDDWG2xq._1UOaKCATq4VGq5WIs6BsnCl9AiN(ThreadStart )
```

**Comment by Craluminum2413** [comment]
Another crash, but for placed blocks like metal spikes
```Running on 64 bit Windows with 16 GB RAM 
Version: v1.16.0-pre.4 (Unstable)12/04/2021 22:58:48: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.CollectibleBehaviorHandbooked.GetHandbookInfo(ItemSlot inSlot, ICoreClientAPI capi, ItemStack[] allStacks, ActionConsumable`1 openDetailPageFor)
   at Vintagestory.GameContent.GuiHandbookItemStackPage.GetPageText(ICoreClientAPI capi, ItemStack[] allStacks, ActionConsumable`1 openDetailPageFor)
   at Vintagestory.GameContent.GuiDialogHandbook.initDetailGui()
   at Vintagestory.GameContent.GuiDialogHandbook.OpenDetailPageFor(String pageCode)
   at Vintagestory.GameContent.ModSystemHandbook.OnHelpHotkey(KeyCombination key)
   at _akZo30aPlEZvmZvWjYHKUYmwhyG._1bFHdMKuMpQ0d5oA500TXqgT5bb(KeyEvent , IWorldAccessor , IPlayer , Boolean , Boolean , Boolean )
   at Vintagestory.Client.NoObf.ClientMain.OnKeyDown(KeyEvent args)
   at Vintagestory.Client.NoObf.ClientPlatformWindows.game_KeyDown(Object sender, KeyboardKeyEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.Platform.NativeWindowBase.OnKeyDown(Key key, Boolean repeat) in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\Platform\NativeWindowBase.cs:line 129
   at OpenTK.Platform.Windows.WinGLNative.HandleKeyboard(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam) in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 639
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam) in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 828
   at OpenTK.Platform.Windows.Functions.DispatchMessage(MSG& msg)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents() in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 1554
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\VintageStory\opentk\src\OpenTK\GameWindow.cs:line 368
   at _0nHMj3ZyzuUD5RQAfs2MzHe55tl._RPVo3GRPmOwZIPP2T9xyc693xVf(_bB0LepbgyiNByBczs3ZgA7xZ3yp , String[] )
   at _7EQaQlsWtbAF9t3X3tTsUlfNlG1._RPVo3GRPmOwZIPP2T9xyc693xVf(ThreadStart )```

**Comment by hlangeveld** [comment]
Reproduces in -rc2

**Comment by hlangeveld** [comment]
And confirmed in -rc3

**Comment by Craluminum2413** [comment]
Fixed in [1.16.0-rc.4](https://www.vintagestory.at/blog.html/news/homesteading-part-2-4th-release-candidate-1160-rc4-r305/#:~:text=Fixed%3A%20Crashes%20related%20to%20handbook%20and%20itemstack%20hover%20debug%20info)
